servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -2782,10 +2784,28 @@ public RoleList getRoleList(ResourceContext ctx, String domainName, Integer limi
             }
         }
 
+        // filter roles by their tags
```

Comment:
In this method when we ask for roles without any members, we generate a new list of roles. In that section we need to update to return the tags as well.